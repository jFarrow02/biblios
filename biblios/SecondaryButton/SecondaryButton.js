"use strict";Object.defineProperty(exports,"__esModule",{value:true});exports["default"]=void 0;var _react=_interopRequireDefault(require("react"));var _propTypes=_interopRequireDefault(require("prop-types"));var _styledComponents=_interopRequireDefault(require("styled-components"));var _styles=_interopRequireDefault(require("../styles"));function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{"default":obj}}var SecondaryButton=function SecondaryButton(props){var children=props.children,_onClick=props.onClick,disabled=props.disabled;var StyledButton=_styledComponents["default"].button.withConfig({displayName:"SecondaryButton__StyledButton",componentId:"sc-16t9p3t-0"})(["padding:15px 25px;border:1px solid ",";border-radius:20px;background:transparent;cursor:",";color:",";font-size:18px;margin:10px 15px 10px 0px;"],function(props){return _styles["default"].getType(props.type)},function(props){return props.disabled?"normal":"pointer"},function(props){return _styles["default"].getType(props.type)});return _react["default"].createElement(StyledButton,{disabled:disabled,onClick:function onClick(){_onClick()},type:props.type},children)};var _default=SecondaryButton;exports["default"]=_default;SecondaryButton.propTypes={children:_propTypes["default"].oneOfType([_propTypes["default"].string,_propTypes["default"].element]),disabled:_propTypes["default"].bool,onClick:_propTypes["default"].func,type:_propTypes["default"].oneOf(["normal","caution","alert","warning","disabled"])};SecondaryButton.defaultProps={children:"Button",disabled:false,onClick:null,type:"normal"};