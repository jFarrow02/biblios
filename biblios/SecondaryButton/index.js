"use strict";Object.defineProperty(exports,"__esModule",{value:true});exports["default"]=void 0;var _SecondaryButton=_interopRequireDefault(require("./SecondaryButton"));function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{"default":obj}}var _default=_SecondaryButton["default"];exports["default"]=_default;