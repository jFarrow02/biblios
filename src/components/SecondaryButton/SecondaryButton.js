import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import stylesConfig from '../../styles';

const SecondaryButton = props =>{
    const { children, onClick, disabled } = props;
    const StyledButton = styled.button`
        padding: 15px 25px;
        border: 1px solid ${ props => { return stylesConfig.getType(props.type) }};
        border-radius: 20px;
        background: transparent;
        cursor: ${props => props.disabled ? 'normal' : 'pointer'};
        color: ${ props => {return stylesConfig.getType(props.type) }};
        font-size: 18px;
    `;
    return(
        <StyledButton
            disabled={disabled}
            onClick={()=>{onClick()}}
            type={props.type}
        >
            {children}
        </StyledButton>
    );
}

export default SecondaryButton;

SecondaryButton.propTypes = {
    children: PropTypes.oneOfType(
        [
            PropTypes.string,
            PropTypes.element,
        ],
    ),
    disabled: PropTypes.bool,
    onClick: PropTypes.func,
    type: PropTypes.oneOf(
        [
            'normal',
            'caution',
            'alert',
            'warning',
            'disabled',
        ]
    ),
};

SecondaryButton.defaultProps = {
    children: 'Button',
    disabled: false,
    onClick: null,
    type: 'normal',
};