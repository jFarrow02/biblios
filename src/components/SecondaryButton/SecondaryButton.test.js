import React from 'react';
import SecondaryButton from './SecondaryButton';
import renderer from 'react-test-renderer';

it('should match a snapshot', ()=>{
    const tree = renderer.create(<SecondaryButton/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});