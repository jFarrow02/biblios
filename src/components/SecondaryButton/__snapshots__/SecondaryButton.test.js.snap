// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match a snapshot 1`] = `
<button
  className="SecondaryButton__StyledButton-sc-16t9p3t-0 eWdKnT"
  disabled={false}
  onClick={[Function]}
  type="normal"
>
  Button
</button>
`;
